src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/EventListener.java

```diff
@@ -332,4 +378,113 @@ public int hashCode() {
     public boolean equals(Object obj) {
         return obj instanceof EventListener && ((EventListener)obj).job.equals(job);
     }
+
+    /**
+     * Inspects {@link ParametersAction} to see what kind of capabilities it has in regards to SECURITY-170.
+     * Assuming the safeParameters constructor could not be found.
+     *
+     * @return the inspection result
+     * @see #createParameters(GerritTriggeredEvent, Job)
+     */
+    private static synchronized ParametersActionInspection getParametersInspection() {
+        if (parametersInspectionCache == null) {
+            parametersInspectionCache = new ParametersActionInspection();
+        }
+        return parametersInspectionCache;
+    }
+
+    /**
+     * Stored cache of the inspection.
+     * @see #getParametersInspection()
+     */
+    private static volatile ParametersActionInspection parametersInspectionCache = null;
+
+    /**
+     * Data structure with information regarding what kind of capabilities {@link ParametersAction} has.
+     * @see #getParametersInspection()
+     * @see #createParameters(GerritTriggeredEvent, Job)
+     */
+    private static class ParametersActionInspection {
```

Comment:
would even make sense to make as separate plugin because all other trigger plugins needs to do the same with same quality level of checks.